### PR TITLE
Add model prediction API

### PR DESCRIPTION
**Pull request by aleksmitov**
Introduce a `Model.predict(input_df: pandas.Dataframe) -> pandas.Dataframe` API.
Implemented for sci-kit learn and catboost models.

### CODE REVIEW

File: layer/flavors/base.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 
+import pandas as pd
 from layerapi.api.entity.model_version_pb2 import ModelVersion
 
 from layer.types import ModelArtifact
 
 
+@dataclass(frozen=True)
+class ModelRuntimeObjects:
+    model_artifact: ModelArtifact
```
Comment (eminn) [inline layer/flavors/base.py:15]: maybe rename to model_object ? and rename the type as well

File: layer/flavors/catboost.py [hunk 2]
```diff
@@ -20,7 +22,15 @@ def save_model_to_directory(
...
         import mlflow.catboost
 
-        return mlflow.catboost.load_model(directory.as_uri())
+        model = mlflow.catboost.load_model(directory.as_uri())
+        return ModelRuntimeObjects(
+            model, lambda input_df: self.__predict(model, input_df)
+        )
+
+    @staticmethod
+    def __predict(model: Any, input_df: pd.DataFrame) -> pd.DataFrame:
```
Comment (eminn) [inline layer/flavors/catboost.py:34]: why do we have underscores we do expect our plugins to call this method.

Comment (aleksmitov) [reply layer/flavors/catboost.py:34]: The plugin will call Model.predict(), i.e the object we return from layer.get_model(), so it's not going to be this one.

Comment (aleksmitov) [reply layer/flavors/catboost.py:34]: I'm talking about `layer/contracts/models.py.`predict() in this PR.

File: layer/flavors/base.py (hunk 1 above)
Comment (aleksmitov) [reply layer/flavors/base.py]: I've renamed it.

File: layer/flavors/catboost.py (hunk 2 above)
Comment (eminn) [reply layer/flavors/catboost.py:34]: I see.  Further comments below:
- We can also change model parameter type to `ModelObject `
- Do we need this method to be static ?

Comment (aleksmitov) [reply layer/flavors/catboost.py]: - Updated the code based on your first point.
- About the second, I think it should remain a static method. I don't see why it should be reading/writing state from the model flavor class instance. The other option is changing the model runtime object that has been loaded with mlflow. I don't think we should do that either.

**Review by eminn — APPROVED** [review]

**Pull request merged**